### PR TITLE
Add User annotation for arguments in userprefs.php

### PR DESCRIPTION
**Pull request by jchaffraix**
In order to have proper PHPStan coverage, first
annotated the return value from User::load_current()
then added User type annotation to userprefs.php.

This uncovered 2 missing properties on Users (`profile`
and `navbar_activity_menu`) and some fields on `UserProfile`
that were typed as `bool` but should have been `int`.

Manually tested updating a user's prefs and changing
the layout during proofing.

### CODE REVIEW

File: tools/proofers/processtext.php [hunk 1]
```diff
@@ -347,7 +347,7 @@
 function switch_layout()
 {
     $user = User::load_current();
-    $user->profile->i_layout = $user->profile->i_layout == 1 ? 0 : 1;
+    $user->profile->i_layout = $user->profile->i_layout == true ? false : true;
```
Comment (cpeel) [inline tools/proofers/processtext.php:350]: This is a good find, but I think the error is in the UserProfile type specification for this, I think the following types are wrong:
```
 * @property bool $i_res;  # should be an int, values are in pinc/resolution.inc
 * @property bool $i_type;  # should be an int -- yes the values are just 0 and 1 but that's really a 2-value enum
 * @property bool $i_layout;  # same about a 2-value enum
 
 * @property bool $v_fntf;  # should be an int -- multi-valued enum
 * @property bool $h_fntf;  # same as ^^

```

**Review by cpeel — COMMENTED** [review]
Thanks for the PR! These kinds of improvements are much appreciated.

File: tools/proofers/processtext.php (hunk 1 above)
Comment (jchaffraix) [reply tools/proofers/processtext.php]: Sounds good, I have applied the fix to the suggested types and reverted the int-to-bool change.

**Comment by cpeel** [comment]
@srjfoo at al, I spun up a test sandbox for this branch at https://www.pgdp.org/~cpeel/c.branch/julien_add_user_in_userprefs/

**Review by cpeel — APPROVED** [review]

**Review by srjfoo — APPROVED** [review]

**Review by 70ray — APPROVED** [review]

**Pull request merged**